utils/fs-operations.js

```diff
@@ -52,10 +52,20 @@ const initStudentDir = (studentDirPath, counterFilePath) => {
 /**
  * @summary Get the value of a counter as a string
  * @function
+ * @throws Throws an error if the counter file is invalid
  * @param {string} counterFilePath
  * @returns {string}
  */
-const getCounter = counterFilePath => fs.readFileSync(counterFilePath).toString().replace('\n', '');
+const getCounter = (counterFilePath) => {
+  // contents of counter file should be only digits followed by a newline
+  const counterRegExp = /^\d+$/;
+  const contents = fs.readFileSync(counterFilePath).toString().replace('\n', '');
```

Comment:
Since you've already had `counterRegExp` for validating the content of counter, could you utilize it to fetch the counter as well so that you don't need to trim the output manually?